src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -735,7 +745,15 @@ public float getFloatValue() throws IOException
             _reportError("Numeric value ("+getText()+") out of range of Java float");
         }
         */
-        return (float) value;
+        if ((_numTypesValid & NR_FLOAT) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_FLOAT);
+            }
+            if ((_numTypesValid & NR_FLOAT) == 0) {
+                convertNumberToFloat();
```

Comment:
Is this called from anywhere? Or is it meant to be called by databind and/or Scala module?

Also:

Reply:
tbh much of the change is just me copy/pasting the NR_DOUBLE bits and having NR_FLOAT bits that match - the `_parseSlowFloat` change is the one that fixes the broken test case - it was still broken till I made that additional change - some of the other changes that were made in this PR prior to that may not be necessary